### PR TITLE
Add two new options (--ignore-var-run and env_file), review the dry-run logic and try to fix some issues

### PR DESCRIPTION
**Pull request by nicolas-r**
Try to fix the following issues
#16 
#17 
#18 
#19

### CODE REVIEW

**Comment by nicolas-r** [comment]
I'll review the comments in "ci/woodpecker/pr/test-release — Pipeline failed" to fix them

**Comment by nicolas-r** [comment]
Hi
You are right, I was too happy that this plugin is updated and I made this PR too quickly
I'm fixing the last issues in my branch and will update this PR

**Comment by nicolas-r** [comment]
Hello

I have clean the code and fix shellchek warnings. But I have to disable some of them as I was not able to find a way to keep them enable with busybox without making Kaniko stop working.

I'll let you review and share your comments

File: plugin.sh [hunk 1]
```diff
@@ -1,11 +1,32 @@
-#!/busybox/sh
+#!/busybox/busybox sh
```
Comment (6543) [inline plugin.sh:1]: Was the exec bit intentionally removed?

Comment (nicolas-r) [reply plugin.sh:1]: I don’t have removed the exec bit, I used this shell to help shellcheck to identify correctly the shell features

**Comment by maltegrosse** [comment]
@nicolas-r great work with your PR. I wasnt able to get this plugin working caus of tags issues (it always chooses "latest" tag, even other tags were setup properly: https://github.com/woodpecker-ci/plugin-kaniko/blob/main/plugin.sh#L95)

Manually injected your plugin.sh and it works fluent. 

Thank you for your efforts and hopefully see soon a new plugin release

**Review by 6543 — APPROVED** [review]

**Pull request merged**